src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -1317,6 +1317,15 @@ public int size() {
                 public Iterator<Map.Entry<K,V>> iterator() {
                     return new MapNIterator();
                 }
+
+                // all mutating methods throw UnsupportedOperationException
```

Comment:
Can we change this anonymous class to `return new AbstractImmutableSet<>` and override `hashCode` with `MapN.this.hashCode()`?

Reply:
Thanks, I tried that out and it looks good based on the test coverage. I have tentatively updated the PR to use that approach.